designs/modify-resources-attached-to-code-pipeline.md

```diff
@@ -0,0 +1,163 @@
+* the notification message has `Yes` and `No` buttons
+
+Proposed:
+* If the lambda function is not associated with a CodePipeline, the same experience takes place.
+* If the lambda function is associated with a CodePipeline, the user is given an alternate prompt (see Confirmation Prompt below). If the user agrees to proceed with function deletion, they do not get an additional prompt. The function is deleted.
+
+### Deleting a CloudFormation Stack
+
+Users delete a CloudFormation Stack through the AWS Explorer by right-clicking on the Stack and selecting `Delete CloudFormation Stack`.
+
+Currently:
+* a notification message pops up with the message "Are you sure you want to delete '`STACK_NAME_HERE`'?"
+* the notification message has `Yes` and `No` buttons
+
+Proposed:
+* If the CloudFormation Stack is not associated with a CodePipeline, the same experience takes place.
+* If the CloudFormation Stack is associated with a CodePipeline, the user is given an alternate prompt (see [Confirmation Prompt](#confirmation-prompt) below). If the user agrees to proceed with Stack deletion, they do not get an additional prompt. The Stack is deleted.
+
+### Deploying a SAM Application to an existing CloudFormation Stack
+
+Users deploy a SAM Application using a Command Palette action. Through a Wizard, they are asked for details about what to deploy, and where to deploy it, including specifying a CloudFormation Stack. The stack specified can be new or may already exist. At this time, users must type the stack name in manually. There is a backlog task to replace this with a list that the user can choose a stack from.
+
+Proposed:
+After the wizard has collected all information from the user, we can check to see if the Stack belongs to a CodePipeline. If it does not, the deploy proceeds. If it does, the user is given a confirmation prompt (see [Confirmation Prompt](#confirmation-prompt) below). If the user agrees to proceed with the deploy, it proceeds normally. Otherwise, nothing happens - the wizard has already ended, and the user is back in the editor.
+
+Alternate:
+After the user enters a CloudFormation Stack, but before the wizard closes, we can check to see if the Stack belongs to a CodePipeline. If it does not, the wizard continues. If it does, the user is given a confirmation prompt, however this confirmation prompt is integrated with the wizard. If the user agrees to proceed with the deploy, the wizard moves on to the next step. Otherwise, the wizard moves back to the step where the user can specify a different CloudFormation Stack.
+
+### <a id="confirmation-prompt"></a>Confirmation Prompt
+
+The confirmation prompt contains the following messaging. See [Success criteria](#success-criteria) for the required information.
+
+---
+
+{ AWS Lambda Function | AWS CloudFormation Stack } '`arn_here`' is part of an AWS CodePipeline pipeline (`pipeline_arn_here`).
```

Comment:
ARNs or friendly names? We don't explicitly show ARNs in any of our existing tooling (you have to hover for a tooltip or get the configuration (Lambda-specific)).

Reply:
The JetBrains Toolkit plans on using ARNs.

The Lambda function and CloudFormation stack could reasonably be friendly name, but the Pipeline should be ARN since we do not surface Pipelines anywhere in the toolkit.

Lets discuss in the design review meeting.

Reply:
The resource of interest will use the friendly name, and the pipeline will use the value stored in the tag, which is likely an arn

Reply:
JetBrains Toolkit also uses friendly names for resource and the pipeline uses the value stored in the tag which is an ARN.